psd-web/spec/models/user_spec.rb

```diff
@@ -417,4 +419,81 @@
       end
     end
   end
+
+  describe "devise gems echosystem" do
+    describe "#send_two_factor_authentication_code", :with_test_queue_adpater do
+      subject(:user) { create(:user) }
+
+      # rubocop:disable RSpec/MultipleExpectations, RSpec/ExampleLength
+      it "enqueues a SendTwoFactorAuthenticationJob" do
+        ActiveJob::Base.queue_adapter = :test
+        expect { user.send_new_otp }
+          .to enqueue_job(SendTwoFactorAuthenticationJob)
+                .at(:no_wait)
+                .on_queue("psd")
+                .with do |user_param, code_param|
+          expect(user_param).to be user
+          expect(user.direct_otp).to eq(code_param)
+        end
+      end
+      # rubocop:enable RSpec/MultipleExpectations, RSpec/ExampleLength
+    end
+
+    describe "#two_factor_lock_expired?" do
+      it "defaults to true when there was no two factor lock for the user" do
+        user = build_stubbed(:user, second_factor_attempts_locked_at: nil)
+        expect(user.two_factor_lock_expired?).to eq true
+      end
+
+      it "returns true when the lock time expired for the user" do
+        expired_lock_time = Time.current - (User::TWO_FACTOR_LOCK_TIME + 10)
```

Comment:
10 hours? 10 seconds? 
```suggestion
        expired_lock_time = (User::TWO_FACTOR_LOCK_TIME + 10.seconds).ago
```

Reply:
Much clearer :clap:

Reply:
Ended taking a `travel_to` approach to make more obvious that the test is happening before/after the locked time expires. (Or at least I found it more readable).